test/int/wizards/pausedWizard.ts

```diff
@@ -0,0 +1,159 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { DebugProtocol } from 'vscode-debugprotocol';
+import { ExtendedDebugClient, THREAD_ID } from 'vscode-chrome-debug-core-testsupport';
+import { logger } from 'vscode-debugadapter';
+import { utils } from 'vscode-chrome-debug-core';
+import { isThisV2 } from '../testSetup';
+import { waitUntilReadyWithTimeout } from '../utils/waitUntilReadyWithTimeout';
+import { expect } from 'chai';
+import { ValidatedMap } from '../core-v2/chrome/collections/validatedMap';
+import { wrapWithMethodLogger } from '../core-v2/chrome/logging/methodsCalledLogger';
+
+enum EventToConsume {
+    Paused,
+    Resumed,
+    None
+}
+
+/** Helper methods to wait and/or verify when the debuggee was paused for any kind of pause.
+ *
+ * Warning: Needs to be created before the debuggee is launched to capture all events and avoid race conditions
+ */
+export class PausedWizard {
+    private _noMoreEventsExpected = false;
+    private _eventsToBeConsumed: (DebugProtocol.ContinuedEvent | DebugProtocol.StoppedEvent)[] = [];
+    private static _clientToPausedWizard = new ValidatedMap<ExtendedDebugClient, PausedWizard>();
+
+    private constructor(private readonly _client: ExtendedDebugClient) {
+        this._client.on('stopped', stopped => {
+            this.validateNoMoreEventsIfSet(stopped);
+            this._eventsToBeConsumed.push(stopped);
+            this.logState();
+        });
+        this._client.on('continued', continued => {
```

Comment:
Looks like these two callbacks are identical, probably can be merged into one.

Reply:
Good idea!